scripts/train/train.py

```diff
@@ -59,6 +59,32 @@ def validate_config(cfg):
                 'ICL evaluation does not currently support Encoder-Decoder models, such as "hf_t5".'
             )
 
+    if (cfg.model.get('fc_type', 'torch') != 'te' and 'te' not in cfg.model.get(
+            'ffn_config', {}).get('ffn_type', 'mptmlp') and
+            'fp8' in cfg.precision):
+        warnings.warn(
+            "fp8 only supported for te.Linear layers. Either set `cfg.model.fc_typ='te'` or "
+            "`cfg.model.ffn_config.ffn_type='te_ln_mlp'` to enable layers using fp8 precision."
+        )
+
+    if ((cfg.model.get('fc_type', 'torch') == 'te' or 'te' not in cfg.model.get(
```

Comment:
these `if` conditions are a little hard to parse quickly because of the defaults. Would it make sense to make to do several levels of `if` statements + a quick comment, rather than as a single `if` line?

Reply:
I split the conditional.

re comment
I feel like the warning is comment enough
```
            warnings.warn(
                '`te.Linear` layers do not support activation_checkpointing with '
                '`activation_checkpointing_reentrant = False`. '
                'Setting cfg.fsdp_config.activation_checkpointing_reentrant=True.'
            )
```
Is there anything else you'd add to it?

Reply:
Yea that should be clear enough. I think the defaults make it harder to parse, but they are necessary for bwd compatibility/ability to not have to fully specify everything so it should be fine. Thanks!